tests/test_routing.py

```diff
@@ -746,3 +758,93 @@ def __call__(self, request):
 )
 def test_route_name(endpoint: typing.Callable, expected_name: str):
     assert Route(path="/", endpoint=endpoint).name == expected_name
+
+
+def assert_middleware_header_route(request: Request):
+    assert getattr(request.state, "middleware_touched") == "Set by middleware"
+    return Response()
+
+
+class AddHeadersMiddleware(BaseHTTPMiddleware):
```

Comment:
Can we avoid the `BaseHTTPMiddleware` on the tests? 👀

Reply:
Yeah good call. I'll change it.

Reply:
Donzo!